delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resolvers/ResolverContextResolution.scala

```diff
@@ -0,0 +1,71 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolutionError.RemoteContextNotAccessible
+import ch.epfl.bluebrain.nexus.delta.sdk.model.ResourceRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Caller
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resources.Resource
+import ch.epfl.bluebrain.nexus.delta.sdk.{Acls, Resolve, Resolvers, ResourceResolution, Resources}
+import io.circe.syntax._
+
+/**
+  * Allows to resolve contexts first via a predefined context resolution and fallback
+  * on a second based on resource resolving in the given project
+  *
+  * @param rcr the static resource resolution
+  * @param resolveResource a function to resolve resources
+  */
+final class ResolverContextResolution(
+    rcr: RemoteContextResolution,
+    resolveResource: Resolve[Resource]
+) {
+
+  def apply(projectRef: ProjectRef)(implicit caller: Caller): RemoteContextResolution =
+    (iri: Iri) =>
+      rcr
+        .resolve(iri)
+        .onErrorFallbackTo(
+          resolveResource(ResourceRef(iri), projectRef, caller)
+            .bimap(
+              report =>
+                RemoteContextNotAccessible(
+                  iri,
+                  s"Resolution via static resolution and via resolvers failed in $projectRef",
+                  Some(report.asJson)
+                ),
+              result => result.compacted.ctx.contextObj.asJson
```

Comment:
I think you should use the source, instead of the compacted (more efficient)

```scala
result.source.topContextValueOrEmpty.contextObj.asJson
```

Reply:
Ok, I thought that was one of the pros of having the compacted resource.